Newsletters: add 403 (2026-05-01) (#2736)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2026-05-01-newsletter.md

  a nonce (or the secp256k1 key) into scripts spent using the SPHINCS key to
  retain privacy equivalent to BIP32 wallets. A benefit of this design choice
  is that the expensive full SPHINCS key derivation can be deferred to the
  first non-hardened derivation step and then cached for all non-hardened keys
  below that step. This wallet design is intended to be combined with
  [BIP360][] P2MR outputs and a future `OP_CHECKSPHINCS` (or similar) to
  enable migration to quantum resistant wallets. Conduition suggests that such
  a wallet structure might also be combined with future lower cost
  post-quantum signature algorithms with SPHINCS providing a dependable
  fallback in case they are proven insecure.

- **In defense of a PQ output type**: Antoine Poinsot [wrote][ap ml pqout] to
  the Bitcoin-Dev mailing list defending a plain post-quantum output type (as
  opposed to a [P2TR][topic taproot]-like output type which allows
  quantum-vulnerable key spending to be disabled by a later soft fork). The
  crux of the argument is that the decision of whether or when it makes sense
  to disable quantum-vulnerable spends should be separated from enabling users

>   to migrate to post-quantum cryptography at their discretion. In the
>   subsequent conversation, some agreement was reached on both adding
>   post-quantum signing to tapscript and adding a plain post-quantum output
>   type. Several open questions remain, including whether and to what degree to

Reviewer: It may make sense to clarify who agreed here:

```suggestion
  to migrate to post-quantum cryptography at their discretion. In the
  subsequent conversation, the participants agreed on both adding
  post-quantum signing to tapscript and adding a plain post-quantum output
  type. Several open questions remain, including whether and to what degree to
```

  Because [BIP342][] signature checking opcodes treat all non-32-byte keys as
  unknown key types which are valid with any non-empty signature, other key
  lengths (in this case with an initial tag byte) can be used in scripts today
  as long as either the scripts are kept secret or they also require a secure
  [BIP340][] signature in addition to the unknown key or keys. If Buchner's
  proposal were to be standardized, wallets could start building scripts with
  various post-quantum key types now while continuing to spend using
  quantum-vulnerable keys until such time as a soft fork enables secure
  spending with the post-quantum keys. Like many quantum migration proposals,
  this proposal only retains security in the face of a quantum adversary if
  key reuse is strictly prevented. Buchner is seeking feedback on the
  proposal.

- **Consensus Cleanup: demo of slow blocks on Signet**: On Delving Bitcoin,
  Antoine Poinsot [wrote][ap delving slowblocks] about a demonstration of the
  types of slow to validate blocks that part of [BIP54][]
  ([consensus cleanup][topic consensus cleanup]) prevents. Over nearly a full

>   day, 3 batches of slow to validated blocks were signed on the most popular

Reviewer: How about:

```suggestion
  ([consensus cleanup][topic consensus cleanup]) prevents. Repeated three times over the course of a
  day, batches of slow to validated blocks were signed on the most popular
```

>   day, 3 batches of slow to validated blocks were signed on the most popular

Reviewer: slow to validated ↦ slow to validate

```suggestion
  day, 3 batches of slow to validate blocks were signed on the most popular
```

  Bitcoin [signet][topic signet] and then reorged away to enable testing of
  propagation and validation behavior of these blocks without forever slowing
  signet initial block download. Many around the world watched the slow blocks
  hit their nodes and logged the validation and propagation behavior. As
  expected, the slow to validate blocks propagated much more slowly through
  the network and required significantly more time to be fully validated on
  individual nodes compared to typical blocks. It should be noted that these
  demonstration blocks were far from the worst case that is prevented by
  BIP54.

- **Post-Quantum BIP-86 Recovery via zk-STARK Proof of BIP-32 Seed Knowledge**:

>   Olauluwa Osuntokun (roasbeef) [posted][oo ml pqrecovery] on the Bitcoin-Dev

Reviewer: ```suggestion
  Olaoluwa Osuntokun (roasbeef) [posted][oo ml pqrecovery] on the Bitcoin-Dev
```

  removed in Bitcoin Core 29.0. The `getrawmempool`, `getmempoolentry`,
  `getmempoolancestors`, and `getmempooldescendants` RPCs no longer return the
  deprecated `bip125-replaceable` field described in [BIP125][] by default.

- [HWI #831][] adds support for the Ledger Nano Gen5 hardware signing device.

- [BDK #2188][] starts verifying that a transaction returned by an Electrum
  server matches the requested txid before caching or using it. Previously, a
  server could respond to a `fetch_tx()` request with any transaction data and a
  different txid, and BDK would accept it.

- [BDK #2115][] adds previous-block-hash awareness to `CheckPoint` by extending
  the `ToBlockHash` trait with an optional `prev_blockhash()` method. This
  allows BDK to verify that adjacent checkpoints connect when their payloads
  contain previous-block-hash information, such as in block headers. This also
  prevents `merge_chains()` from treating a conflicting height-0 checkpoint as a
  normal reorg and replacing it. Now, if two checkpoint chains disagree on
  genesis, the merge fails. See Newsletters [#372][news372 checkpoint] and
  [#390][news390 checkpoint] for previous work on `CheckPoint`.

> - [BIPs #1548][] adds [BIP391][], a specification for Binary Output Descriptors

Reviewer: Reminder to reorder

Reviewer 2: thx , done
